src/ui/screens/Council/CouncilScreen.tsx

```diff
@@ -283,6 +305,101 @@ function CouncilVoteModal({visible, setVisible, candidates, moduleElection}: Cou
   );
 }
 
+function SubmitCandidancyModel({visible, setVisible, moduleElection}: SubmitCandidancyProps) {
+  const [account, setAccount] = React.useState<Account>();
+  const startTx = useApiTx();
+  const {api} = useApi();
+  const balance = useFormatBalance();
+  const formattedBalance = balance.formatBalance(moduleElection.candidacyBond);
+  const {data: council} = useCouncil();
+  const {colors} = useTheme();
+  const snackbar = useSnackbar();
+  const onSubmitCandidacy = () => {
+    console.log(moduleElection.module);
+    if (account && api && moduleElection.module) {
+      startTx({
+        address: account.address,
+        txMethod: `${moduleElection.module}.submitCandidacy`,
+        params: [
+          api.tx[moduleElection.module]?.submitCandidacy?.meta.args.length === 1 ? [council?.candidates.length] : [],
+        ],
+      })
+        .then((tx) => {
+          snackbar('Candidacy submitted successfully');
+        })
+        .catch((tx) => {
+          snackbar('Error while submitting candidacy');
+        });
+      reset();
+    }
+  };
+
+  const reset = () => {
+    setAccount(undefined);
+    setVisible(false);
+  };
+
+  const accountPopableProps = {
+    content: 'Select the account you wish to submit for candidacy.',
+    children: null,
+    backgroundColor: colors.accent,
+  };
+
+  const candidancyBondPopableProps = {
+    content: 'Select the account you wish to submit for candidacy.',
+    children: null,
+    backgroundColor: colors.accent,
+  };
```

Comment:
If we have to do this every time we need to use the `<Popover />` component that's tedious. Ideally the component should only receive the `popableText: string` as prop and take care of everything else.